packages/blockchain-wallet-v4-frontend/src/services/errors/NabuError/utils/createNabuErrorAnalyticsInterceptor/createNabuErrorAnalyticsInterceptor.ts

```diff
@@ -0,0 +1,29 @@
+import { actions } from 'data'
+import { ClientErrorProperties } from 'data/analytics/types/errors'
+import { Analytics } from 'data/types'
+
+import { isNabuError } from '../isNabuError'
+import { CreateNabuErrorAnalyticsInterceptorUtility } from './createNabuErrorAnalyticsInterceptor.types'
+
+export const createNabuErrorAnalyticsInterceptor: CreateNabuErrorAnalyticsInterceptorUtility =
+  (dispatch) => (error) => {
+    if (isNabuError(error)) {
+      const clientError: ClientErrorProperties = {
+        category: error.categories,
+        error: 'NABU_ERROR',
+        network_error_description: error.message,
+        network_error_type: 'NABU_ERROR',
+        source: 'NABU',
+        title: error.title
+      }
+
+      dispatch(
+        actions.analytics.trackEvent({
+          key: Analytics.CLIENT_ERROR,
+          properties: clientError
+        })
+      )
+    }
+
+    return Promise.reject(error)
```

Comment:
Is this line what would normally happen if this interceptor didn't exist and the api returned a 4xx or 5xx? So if the error is not a nabu error, it won't effect any of our existing error handling?

Reply:
exactly

Reply:
yes, it will not affect the current handling because the error will always go through and the normal handlers should take care of the error